clients/src/main/java/org/apache/kafka/common/record/OldLogEntry.java

```diff
@@ -0,0 +1,412 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ * <p/>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p/>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+package org.apache.kafka.common.record;
+
+import org.apache.kafka.common.KafkaException;
+import org.apache.kafka.common.errors.CorruptRecordException;
+import org.apache.kafka.common.utils.AbstractIterator;
+import org.apache.kafka.common.utils.ByteBufferInputStream;
+import org.apache.kafka.common.utils.Utils;
+
+import java.io.DataInputStream;
+import java.io.DataOutputStream;
+import java.io.EOFException;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.ArrayDeque;
+import java.util.Collections;
+import java.util.Iterator;
+
+import static org.apache.kafka.common.record.Records.LOG_OVERHEAD;
+import static org.apache.kafka.common.record.Records.OFFSET_OFFSET;
+
+public abstract class OldLogEntry extends AbstractLogEntry implements LogRecord {
```

Comment:
Yes, the name is less than ideal. We also need to rename `EosLogEntry` and `EosRecord`. Our code will mostly work through the interfaces, so they need not be perfect, but there is plenty of room for improvement. That said, I expect future message format changes will continue to use the new log entry and record objects, so I'm not too keen to pack version information into the names. Maybe it is ok to have `NewLogEntry` and `OldLogEntry` until we actually have need for a third type? 

The other thing I'm considering is pulling `Record` into `OldLogEntry` to make it clear that it is only for the older message format. What do you think?